pydicom/dataset.py

```diff
@@ -2490,18 +2491,23 @@ def to_json_dict(
             :class:`Dataset` representation based on the DICOM JSON Model.
         """
         json_dataset = {}
-        for key in self.keys():
-            json_key = '{:08X}'.format(key)
-            try:
-                data_element = self[key]
-                json_dataset[json_key] = data_element.to_json_dict(
-                    bulk_data_element_handler=bulk_data_element_handler,
-                    bulk_data_threshold=bulk_data_threshold
-                )
-            except Exception as exc:
-                logger.error(f"Error while processing tag {json_key}")
-                if not suppress_invalid_tags:
-                    raise exc
+        context = (
+            config.strict_reading() if suppress_invalid_tags
+            else nullcontext()
+        )
+        with context:
```

Comment:
Allowing ISfloat by default caused some json tests to fail.  Added this strict context to try to keep previous behavior.

Reply:
I have some trouble understanding the logic here: if `suppress_invalid_tags` is set, we raise an exception, and will be cught and ignored here? I'll probably have another look tomorrow with a fresh brain...

Reply:
I checked the failing test (`test_suppress_invalid_tags_with_failed_dataelement`) - I wrote that, and I think it just should be changed to use another invalid value that would still raise, for example:
```py
        ds[0x00082128] = RawDataElement(
            Tag(0x00082128), 'IS', 4, b'5:25', 0, True, True
        )

        with pytest.warns(UserWarning):
            ds_json = ds.to_json_dict(suppress_invalid_tags=True)
        assert "00082128" not in ds_json
```
and revert the change here.

Reply:
> ...I think it just should be changed to use another invalid value that would still raise

I thought about that too, but that would still change existing behavior for IS - ones that did not get written before would now be accepted and written out.  Probably not a big deal, but still something a bit different.  I'm not sure just skipping values in the output is the best idea anyway.  I'll think about this a bit before this is merged.  I'd still like to have a "bigger picture" in mind for how to handle settings in v3.X...  I'll start an issue and post some thoughts on that in the coming days.

Reply:
> but that would still change existing behavior for IS - ones that did not get written before would now be accepted and written out

That is true, but I assumed that this is a change we want to make. Though maybe I haven't thought this through... I will wait your thoughts then.

Reply:
Having a look at this again ... This change has no effect (`nullcontext()`) for the default case where `suppress_invalid_tags` is False.  So I think I was right, if `suppress...` is True (as seen only in tests probably at the moment), then we will catch the exception and ignore it, as is currently done (the new `ISFloat` is not allowed with `strict_reading()` turned on).

It is still messy, but I'd rather push real fixes to this messy config (and all the other messy config) to v3.0.

Reply:
Well, I leave the decision to you. I find it a bit ugly, but you are probably right - better defer the real fix to 3.0, where behavior changes are expected.